### PR TITLE
Helm discovery changes - Review

### PR DESCRIPTION
**Pull request by paliwalparitosh**

### CODE REVIEW

File: charts/logan/values.yaml [hunk 1]
```diff
@@ -477,3 +488,44 @@ fluentd:
...
+      # oci_la_namespace: Logging Analytics namespace; overrides oci-onm-common.ociLANamespace
+      oci_la_namespace:
+      # oci_la_log_group_id: OCID of Logging Analytics Log Group to send kubernetes object logs to; overrides oci-onm-common.ociLALogGroupID
+      oci_la_log_group_id:
+      # oci_la_log_group_id: Logging Analytics Ingestion API endpoint to ingest your object logs; overrides oci-onm-common.ociLAEndpoint
+      oci_la_endpoint:
+   # kubernetes: kubernetes cluster related inputs for kubernetes disocvery job
+   kubernetes:
+      # kube_config_location: if provided, disocvery job will use the specific kube config file to connect to the cluster
+      kube_config_location:
```
Comment (santhoshkvuda) [inline charts/logan/values.yaml:517]: remove.

File: charts/logan/templates/discovery-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            - name: ociconfigdir
+              mountPath: {{ .Values.oci.path }}
+              readOnly: true
+            {{- end }}
+            command:
+            {{- /* object discovery script */}}
+              - oci-logan-oke-resources-discovery
+            {{- /* mandatory inputs */}}
+              - --kubernetes_cluster_id
+            {{- if ne $kubernetesClusterId "UNDEFINED" }}
```
Comment (santhoshkvuda) [inline charts/logan/templates/discovery-cronjob.yaml:34]: if condition not req. Directly take $kubernetesClusterId

File: charts/logan/templates/discovery-cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+              - {{ $kubernetesClusterId }}
+            {{- else }}
+              {{ required "kubernetes_cluster_id is required" .Values.global.kubernetesClusterID }}
+            {{- end }}              
+              - --kubernetes_cluster_name
+            {{- if ne $kubernetesClusterName "UNDEFINED" }}
```
Comment (santhoshkvuda) [inline charts/logan/templates/discovery-cronjob.yaml:40]: if condition not req. Directly take $kubernetesClusterName

File: charts/logan/templates/discovery-cronjob.yaml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+              - {{ $kubernetesClusterName }}
+            {{- else }}
+              {{ required "kubernetes_cluster_name is required" .Values.global.kubernetesClusterName }}
+            {{- end }}
+              - --oci_la_namespace
+            {{- if .Values.objectDiscovery.logan.oci_la_namespace }}
```
Comment (santhoshkvuda) [inline charts/logan/templates/discovery-cronjob.yaml:46]: remove if, else and mandatory check. just take .Values.ociLANamespace

File: charts/logan/templates/discovery-cronjob.yaml [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+              - {{ .Values.objectDiscovery.logan.oci_la_namespace }}
+            {{- else if .Values.ociLANamespace }}
+              - {{ .Values.ociLANamespace }}
+            {{- else }}
+              {{ required "ociLANamespace is required" .Values.ociLANamespace }}
+            {{- end }}
+            {{- /* optional logan inputs */}}
+            {{- if .Values.objectDiscovery.logan.oci_la_log_group_id }}
```
Comment (santhoshkvuda) [inline charts/logan/templates/discovery-cronjob.yaml:54]: we can remove this for now in the context of discovery only.

File: charts/logan/templates/discovery-cronjob.yaml [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+              - --oci_la_log_group_id
+              - {{ .Values.objectDiscovery.logan.oci_la_log_group_id }}
+            {{- else if .Values.ociLALogGroupID }}
+              - --oci_la_log_group_id
+              - {{ .Values.ociLALogGroupID }}
+            {{- end }}
+            {{- if .Values.ociLAClusterEntityID }}
```
Comment (santhoshkvuda) [inline charts/logan/templates/discovery-cronjob.yaml:61]: we can make this mandate for now in the context of discovery only.

**Review by santhoshkvuda — APPROVED** [review]

**Review by shivamshishodia — APPROVED** [review]

**Review by palukuri — APPROVED** [review]

**Review by pankajtalk — APPROVED** [review]

**Pull request merged**